Enhance RewirePipeline command to support optional pipeline ID vs pipeline name

georgy-gorelko opened the pull request:
### Pipeline ID Support 

- Added new `--ado-pipeline-id `option to support optional numeric pipeline IDs
- Added comprehensive validation logic ensuring mutual exclusivity of pipeline name vs ID parameters
- Added [GetPipelineId] helper method that returns provided ID directly or looks up ID by name
- Updated both dry-run and regular execution paths to use the new resolution logic
- Updated unit tests: Added tests for the new functionality and fixed existing tests to account for the new option

### Key Features Implemented:

1. **Mutual Exclusion Validation**: The command now properly validates that users specify either --ado-pipeline OR --ado-pipeline-id, but not both or neither

1. **Backward Compatibility**: Existing functionality with --ado-pipeline continues to work exactly as before

1. **Enhanced Help Tex**t: The help output clearly explains both options and their usage

1. **Robust Testing**: Added comprehensive unit tests for all validation scenarios and functionality paths

1. **Improved Reliability**: Using pipeline IDs eliminates the need for name-based lookups, making pipeline identification more reliable

Commit:
Enhance RewirePipeline command to support optional pipeline ID and validation logic

danquirk commented on src/ado2gh/Commands/RewirePipeline/RewirePipelineCommandHandler.cs:
```diff
@@ -39,13 +51,25 @@ private async Task HandleRegularRewire(RewirePipelineCommandArgs args)
+        }
+        
+        _log.LogInformation($"Looking up pipeline ID for: {args.AdoPipeline}");
+        return await _adoApi.GetPipelineId(args.AdoOrg, args.AdoTeamProject, args.AdoPipeline);
```
this overload doesn't exist? When it does consider logging the resolved pipeline ID here unless this overload does it itself. Then this overload logs the resolved ID for both input types, ex:
```cs
_log.LogInformation($"Looking up...")
var pipelineId = await _adoApi.GetPipelineId(...)
_log.LogInformation($"Using resolved pipeline ID: ...)
```

Commit:
Log resolved pipeline ID during lookup in RewirePipeline command

Commit:
Log resolved pipeline ID during lookup in RewirePipeline command

Commit:
Merge remote-tracking branch 'origin/ado-pipelines-dry-run' into ado-pipeline-id

georgy-gorelko commented on global.json:
```diff
@@ -1,6 +1,6 @@
 {
   "sdk": {
-    "version": "8.0.410",
+    "version": "8.0.406",
```
unintentional. Needs revert

Commit:
Update SDK version in global.json to 8.0.410

georgy-gorelko committed:
Merge branch 'main' into ado-pipeline-id

georgy-gorelko committed:
Merge branch 'main' into ado-pipeline-id

timisoetan committed:
Add actions langauge support to codeql config

timisoetan committed:
fix

timisoetan committed:
add security-and-quality query

begonaguereca committed:
Merge pull request #1417 from georgy-gorelko/ado-pipelines-dry-run

'gh ado2gh rewire-pipeline': support test rewiring mode before actual rewiring

georgy-gorelko committed:
Merge branch 'main' into ado-pipeline-id

georgy-gorelko committed:
Update RewirePipelineCommandHandler.cs

Commit:
Remove unnecessary blank line at the end of RewirePipelineCommandHandler.cs

Commit:
Revert global.json

timisoetan committed:
Merge pull request #1432 from github/timisoetan/codeql-actions-language-support

Add actions language support to CodeQL analysis

georgy-gorelko committed:
Merge branch 'main' into ado-pipeline-id

Commit:
Enhance error handling in AdoPipelineTriggerService and RewirePipelineCommandHandler

- Added try-catch blocks to handle HTTP request exceptions in AdoPipelineTriggerService.
- Implemented logging for 404 errors and other HTTP errors during pipeline retrieval.
- Updated RewirePipelineCommandHandler to throw exceptions for pipeline not found and lookup failures, with appropriate logging.
- Introduced unit tests for error handling scenarios in both AdoPipelineTriggerService and RewirePipelineCommandHandler.
